kakarot_scripts/setup/setup.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+
+import os
+import shutil
+import subprocess
+import sys
+from typing import Optional
+
+KATANA_VERSION = "v1.0.0-alpha.14"
+PYTHON_MIN_VERSION = (3, 10)
+ASDF_VERSION = "v0.14.1"
+
+SHELL_CONFIG_FILES = {
+    "bash": [".bashrc", ".bash_profile"],
+    "zsh": [".zshrc"],
+    "fish": [".config/fish/config.fish"],
+}
+
+
+class SetupError(Exception):
+    """Custom exception for setup errors."""
+
+    pass
+
+
+def run_command(command: str, error_message: str) -> None:
+    try:
+        subprocess.run(command, check=True, shell=True)
+    except subprocess.CalledProcessError as e:
+        raise SetupError(f"{error_message}: {e}") from e
+
+
+def is_command_available(command: str) -> bool:
+    return shutil.which(command) is not None
+
+
+def get_version(command: str) -> Optional[str]:
+    try:
+        result = subprocess.run([command, "--version"], capture_output=True, text=True)
+        return result.stdout.strip()
+    except subprocess.CalledProcessError:
+        return None
+
+
+def get_shell_config_file() -> Optional[str]:
+    shell = os.environ.get("SHELL", "").split("/")[-1]
+    home = os.path.expanduser("~")
+
+    for file in SHELL_CONFIG_FILES.get(shell, []):
+        full_path = os.path.join(home, file)
+        if os.path.exists(full_path):
+            return full_path
+
+    return None
+
+
+def install_dependency(
+    name: str, install_command: str, check_command: str, version: Optional[str] = None
+) -> None:
+    if is_command_available(check_command):
+        if version:
+            current_version = get_version(check_command)
+            if current_version == version:
+                print(f"{name} version {version} is already installed.")
+                return
+            print(f"Updating {name} to version {version}...")
+        else:
+            print(f"{name} is already installed.")
+            return
+    else:
+        print(f"Installing {name}...")
+
+    run_command(install_command, f"Failed to install/update {name}")
+
+
+def setup_katana() -> None:
+    install_dependency(
+        "katana",
+        f'cargo install --git https://github.com/dojoengine/dojo --locked --tag "{KATANA_VERSION}" katana',
+        "katana",
+        version=KATANA_VERSION,
+    )
+
+
+def setup_local() -> None:
+    if sys.version_info < PYTHON_MIN_VERSION:
+        raise SetupError(
+            f"Python {PYTHON_MIN_VERSION[0]}.{PYTHON_MIN_VERSION[1]} or higher is required."
+        )
```

Comment:
install version with `uv`. This script should be able to run with even the lowest versions